backend/tests/test_models.py

```diff
@@ -18,20 +23,132 @@ def test_delete_lesson_with_exercises(self, db_session, lesson_with_exercise):
         )
         assert not deleted_lesson
 
-    def test_progress_cascade_deleted_for_lesson(self):
+    def test_progress_cascade_deleted_for_lesson(self, db_session):
         """
         Progress records should be deleted if their lesson object was deleted.
         """
-        pass
 
-    def test_progress_cascade_deleted_for_user(self):
+        user = User(
+            id=1,
+            email=faker.email(),
+            background=faker.color(),
+            emoji=faker.color(),
+            password=faker.password(),
+            streak=1,
+        )
+
+        db_session.add(user),
+        db_session.commit(),
+        db_session.refresh(user)
+
+        lesson = Lesson(
+            name=faker.name(),
+            content=faker.text(),
+            preview=faker.url(),
+            order=0,
+        )
+        db_session.add(lesson)
+        db_session.commit()
+        db_session.refresh(lesson)
+
+        progress = Progress(
+            user_id=user.id,
+            lesson_id=lesson.id,
+        )
+
+        db_session.add(progress)
+        db_session.commit()
+
+        assert progress
+
+        db_session.delete(lesson)
+        db_session.commit()
+
+        try:
+            assert not progress.lesson_id
+        except ObjectDeletedError:
+            db_session.delete(user)
+            db_session.commit()
+            return 0
```

Comment:
it fails here, because you attempt to get an attribute of a deleted object. I proposed a different option:
```suggestion
    lesson_progress = db_session.query(Progress).filter(Progress.lesson_id == lesson.id).all()

    assert not lesson_progress
```